tasks/extract-binaries-from-image/extract-binaries-from-image.yaml

```diff
@@ -41,28 +45,43 @@ spec:
             echo "Error: No valid snapshot file was provided."
             exit 1
         fi
-        IMAGE_URL=$(jq -r '.components[0].containerImage // ""' "${SNAPSHOT_SPEC_FILE}")
-        if [ -z "${IMAGE_URL}" ] ; then
-            echo "Error: Unable to get image url from snapshot."
-            exit 1
-        fi
 
         BINARIES_DIR=binaries
         BINARIES_PATH=$(workspaces.data.path)/$(params.subdirectory)/$BINARIES_DIR
         mkdir -p $BINARIES_PATH
 
-        TMP_DIR=$(mktemp -d)
-        skopeo copy docker://$IMAGE_URL dir:$TMP_DIR
+        SNAPSHOT_COMPONENTS_LIST=$(jq -r '.components[] | @base64' "${SNAPSHOT_SPEC_FILE}")
 
-        cd $TMP_DIR
+        # Capture the desired components list from arguments
+        DESIRED_COMPONENTS_LIST="$@"
 
-        for DIGEST in $(jq -r ".layers[].digest" manifest.json)
-        do
-            FILE=${DIGEST#sha256:}
-            tar -xzvf $FILE
-        done
+        for COMPONENT in $SNAPSHOT_COMPONENTS_LIST; do
+          COMPONENT_NAME=$(echo $COMPONENT | base64 --decode | jq -r '.name')
```

Comment:
Instead of calling `base64` every time you need to extract something (`name` and later `containerImage`), you could load the component once and then select what you need. But also, consider using this approach that we use in multiple other tasks:
```sh
NUM_COMPONENTS=$(jq '.components | length' "$SNAPSHOT_SPEC_FILE")
for ((i=0; i < NUM_COMPONENTS; i++)); do
    COMPONENT=$(jq -c ".components[$i]" "$SNAPSHOT_SPEC_FILE")
```

Reply:
Good point, thanks.  I'll refactor this in the next commit.

Reply:
Yeah, I am missing why there is any encoding here at all

Reply:
> Yeah, I am missing why there is any encoding here at all

My understanding was that it was just another way of avoiding the space issue when iterating over json strings via `for` in bash.

Reply:
The encoding wasn't necessary. Removed. thanks

Reply:
The encoding made sure that the strings did not contain any spaces. The current version is not really safe in that regard. So I'd still suggest to iterate over indexes - see my comment at the top of this thread.

Reply:
We actually have a ticket Martin worked on recently to switch all of our loops to looping over indexes (because of spaces), so yeah use that here too please

Reply:
I see. Good suggestion, done.

Reply:
Looks good now, thanks.